core/src/main/java/com/linecorp/armeria/common/HttpRequest.java

```diff
@@ -280,6 +282,55 @@ static HttpRequest of(RequestHeaders headers, Publisher<? extends HttpObject> pu
         }
     }
 
+    /**
+     * Creates a new HTTP request whose {@link Publisher} is produced by the specified
+     * {@link CompletionStage}. If the specified {@link CompletionStage} fails, the returned request will be
+     * closed with the same cause as well.
+     *
+     * @param stage the {@link CompletionStage} which will produce the actual data and trailers
+     */
+    static HttpRequest from(RequestHeaders headers,
```

Comment:
Question) would it rather make more sense to name this method `of`, and create the counterparts in `HttpResponse`? (and deprecate `from()` factory methods)

Reply:
Changed to `of`. :wink:
For `HttpResponse`, I actually left `TODO` for it because it needs to change quite a lot.
https://github.com/line/armeria/blob/83b3b6ca5afdf133f71ff8ff3ed9611a8fa244a8/core/src/main/java/com/linecorp/armeria/common/HttpResponse.java#L88
Let me send another PR for that. 😉